pkgs/development/python-modules/scrapy/default.nix

```
@@ -63,19 +62,28 @@ buildPythonPackage rec {
 
   LC_ALL = "en_US.UTF-8";
 
-  # Disable doctest plugin—enabled in the shipped pytest.ini—because it causes pytest to hang
-  # Ignore proxy tests because requires mitmproxy
-  # Ignore utils_display tests because it requires pygments
-  # Ignore test_retry_dns_error because tries to resolve an invalid dns and weirdly fails with "Reactor was unclean"
-  # Ignore xml encoding test on darwin because lxml can't find encodings https://bugs.launchpad.net/lxml/+bug/707396
-  checkPhase = ''
+  # Disable doctest plugin because it causes pytest to hang
+  preCheck = ''
     substituteInPlace pytest.ini --replace "--doctest-modules" ""
-    pytest --ignore=tests/test_linkextractors_deprecated.py --ignore=tests/test_proxy_connect.py --ignore=tests/test_utils_display.py --deselect tests/test_crawl.py::CrawlTestCase::test_retry_dns_error ${lib.optionalString stdenv.isDarwin "--deselect tests/test_utils_iterators.py::LxmlXmliterTestCase::test_xmliter_encoding"}
   '';
 
+  pytestFlagsArray = [
+    "--ignore=tests/test_proxy_connect.py"
+    "--ignore=tests/test_utils_display.py"
+    "--ignore=tests/test_command_check.py"
+    "--deselect=tests/test_feedexport.py::FTPFeedStorageTest"
+    "--deselect=tests/test_request_left.py::TestCatching::test_noconnect"
+    "--deselect=tests/test_crawl.py::CrawlTestCase::test_retry_dns_error"
```

for individual tests, you can do 
```suggestion
    disabledTests = [ "test_retry_dns_error" ];
```
easier to read than `--deselect={path}::{Module}::{method}`

---

Fixed, thanks!